pkg/costattribution/active_tracker.go

```diff
@@ -169,19 +228,42 @@ func (at *ActiveSeriesTracker) Collect(out chan<- prometheus.Metric) {
 	at.observedMtx.RLock()
 	if !at.overflowSince.IsZero() {
 		var activeSeries int64
+		var nativeHistogram int64
+		var nhBucketNum int64
 		for _, as := range at.observed {
-			activeSeries += as.Load()
+			activeSeries += as.activeSeries.Load()
+			nativeHistogram += as.nativeHistograms.Load()
+			nhBucketNum += as.nativeHistogramBuckets.Load()
 		}
 		at.observedMtx.RUnlock()
-		out <- prometheus.MustNewConstMetric(at.activeSeriesPerUserAttribution, prometheus.GaugeValue, float64(activeSeries+at.overflowCounter.Load()), at.overflowLabels[:len(at.overflowLabels)-1]...)
+		out <- prometheus.MustNewConstMetric(at.activeSeriesPerUserAttribution, prometheus.GaugeValue, float64(activeSeries+at.overflowCounter.activeSeries.Load()), at.overflowLabels[:len(at.overflowLabels)-1]...)
+
+		if nhcounter := float64(nativeHistogram + at.overflowCounter.nativeHistograms.Load()); nhcounter > 0 {
+			out <- prometheus.MustNewConstMetric(at.activeNativeHistogramSeriesPerUserAttribution, prometheus.GaugeValue, nhcounter, at.overflowLabels[:len(at.overflowLabels)-1]...)
+			if bcounter := float64(nhBucketNum + at.overflowCounter.nativeHistogramBuckets.Load()); bcounter > 0 {
+				out <- prometheus.MustNewConstMetric(at.activeNativeHistogramBucketsPerUserAttribution, prometheus.GaugeValue, bcounter, at.overflowLabels[:len(at.overflowLabels)-1]...)
+			}
+		}
 		return
 	}
 	// We don't know the performance of out receiver, so we don't want to hold the lock for too long
 	var prometheusMetrics []prometheus.Metric
 	for key, as := range at.observed {
 		keys := strings.Split(key, string(sep))
 		keys = append(keys, at.userID)
-		prometheusMetrics = append(prometheusMetrics, prometheus.MustNewConstMetric(at.activeSeriesPerUserAttribution, prometheus.GaugeValue, float64(as.Load()), keys...))
+		prometheusMetrics = append(prometheusMetrics,
+			prometheus.MustNewConstMetric(at.activeSeriesPerUserAttribution, prometheus.GaugeValue, float64(as.activeSeries.Load()), keys...),
+		)
+		if as.nativeHistograms.Load() > 0 {
+			prometheusMetrics = append(prometheusMetrics,
+				prometheus.MustNewConstMetric(at.activeNativeHistogramSeriesPerUserAttribution, prometheus.GaugeValue, float64(as.nativeHistograms.Load()), keys...),
+			)
+			if as.nativeHistogramBuckets.Load() > 0 {
```

Comment:
same as above, load into local variable bcounter

Reply:
addressed in 54875a7a10319987325308faf27fbe58129527f2